doc/development/rfc/version_numbering.md

```diff
@@ -0,0 +1,146 @@
+said _preview_ anyway because the odd number does not indicate the
+development version.
+Hence, odd numbers for development versions do not bring any advantage.
+
+The branching and release procedure with even minor numbers for releases
+and odd numbers for development versions require that when new branch for
+a minor version is created, the branch needs to change all the mentions
+of the minor version number to the next even number and, at the same time,
+the _main_ branch needs to change to the next odd number.
+Without the even and odd distinction, the branch keeps the version from
+the main branch while the main branch advanced to the next upcoming version
+resulting only in one operation which is the same whether the current version
+is even or odd.
+
+The Semantic Versioning uses labels after the version number to indicate development
+versions, so using that system or a similar one should be sufficient to mark the
+development versions.
+
+## Version Numbering Specification
+
+### Major
+
+In accordance with Semantic Versioning, major release must happen with
+any backwards incompatible change in the API which includes
+both interface and behavior changes (API also defines what happens,
+not just names and signatures).
+
+Additionally, major feature additions which would require only minor release in terms of API stability,
+are strongly suggested to trigger major release as well.
+This in turn helps to address the issue of outdated tutorials and small, but breaking, changes.
+
+Some major features such as changes in the GUI are fully backwards compatible
+(GUI API is not a public API in versions 7 and 8), but are heavily influencing tutorials and
+other teaching materials. In that sense, behavior linked to a particular interface is changed,
+so a major version is required.
+
+There is always a list of many small changes which are not backwards compatible,
+for example, a cleanup of depreciated functions. None of these changes alone does
+not seem worth a major release, but because it would require one, it is never done.
+With more common major releases, small changes can happen more often.
+
+### Minor
+
+Minor version must be incremented if new functionality is added
+or if existing functionality is marked as deprecated.
+Minor version increment is strongly recommend for all new functionality or improvements
+which are not bug fixes.
+
+### Micro
+
+If only backwards compatible bug fixes, i.e., fixes of incorrect behavior, are applied,
+a micro version can be released.
+
+The micro version is also known as _patch_ which is what Semantic Versioning is using
+and _point_ which is what was used in some GRASS GIS documents. The work _micro_ is used
+to avoid collision with patch in the meaning of individual changeset or fix
+(release contain one or more of these changes). Point release (or _dot_ release)
+can generally apply to anything after the first dot, so minor or micro release.
+
```

Comment:
```suggestion
can generally apply to anything after the first dot, so minor or micro release
is preferred.

```

Reply:
The original wording was unclear. I added more explanation there.